### PR TITLE
feat: DEV-469: Skip queue

### PR DESCRIPTION
**Pull request by triklozoid**

### CODE REVIEW

File: label_studio/tests/skip_queue.tavern.yml [hunk 1]
```diff
@@ -0,0 +1,268 @@
+---
```
Comment (makseq) [inline label_studio/tests/skip_queue.tavern.yml:1]: better to rename skip_queue.tavern.yml to next_task_skip_queue.tavern.yml

**Review by makseq — CHANGES_REQUESTED** [review]

File: label_studio/projects/models.py [hunk 2]
```diff
@@ -84,6 +84,13 @@ def with_counts_annotate(queryset):
 
 
 class Project(ProjectMixin, models.Model):
+    class SkipQueue(models.TextChoices):
```
Comment (makseq) [inline label_studio/projects/models.py:87]: Do we need to put this class here?

Comment (triklozoid) [reply label_studio/projects/models.py:87]: It's for more handy usage via `project.SkipQueue.REQUEUE_FOR_ME` 

Example from django docs: https://docs.djangoproject.com/en/3.0/ref/models/fields/#enumeration-types

File: label_studio/projects/functions/next_task.py [hunk 3]
```diff
@@ -0,0 +1,256 @@
...
+    if project.show_overlap_first:
+        # don't output anything - just filter tasks with overlap
+        logger.debug(f'User={user} tries overlap first from prepared tasks')
+        _, not_solved_tasks = _try_tasks_with_overlap(not_solved_tasks)
+        # queue_info += 'Show overlap first'
+
+    if project.skip_queue == project.SkipQueue.REQUEUE_FOR_ME:
+        # https://code.djangoproject.com/ticket/19726
+        if settings.DJANGO_DB == settings.DJANGO_DB_SQLITE:
+            not_solved_tasks = not_solved_tasks.order_by('annotations__was_cancelled', 'id')
```
Comment (makseq) [inline label_studio/projects/functions/next_task.py:167]: 1. why do we use '-annotations__was_cancelled' for other dbs? 
2. why isn't this covered by tests and line 1?

File: label_studio/projects/functions/next_task.py [hunk 4]
```diff
@@ -0,0 +1,256 @@
...
+    # if annotator is assigned for tasks, he must to solve it regardless of is_labeled=True
+    if not assigned_flag:
+        not_solved_tasks = not_solved_tasks.filter(is_labeled=False)
+
+    # show tasks with overlap > 1 first
+    if project.show_overlap_first:
+        # don't output anything - just filter tasks with overlap
+        logger.debug(f'User={user} tries overlap first from prepared tasks')
+        _, not_solved_tasks = _try_tasks_with_overlap(not_solved_tasks)
+        # queue_info += 'Show overlap first'
```
Comment (makseq) [inline label_studio/projects/functions/next_task.py:162]: Did you commented this? Or it was before

File: label_studio/projects/functions/next_task.py [hunk 5]
```diff
@@ -0,0 +1,256 @@
...
+            f'Uncertainty sampling fallbacks to random sampling '
+            f'(current project.model_version={str(project.model_version)})'
+        )
+        next_task = _get_random_unlocked(tasks, user)
+    return next_task
+
+
+def get_not_solved_tasks_qs(user, project, prepared_tasks, assigned_flag):
+    if project.skip_queue == project.SkipQueue.REQUEUE_FOR_ME:
+        user_solved_tasks_array = user.annotations.filter(ground_truth=False, was_cancelled=False)
```
Comment (makseq) [inline label_studio/projects/functions/next_task.py:145]: Could we add this: 
```
queue_info += ' Requeued for me from cancelled tasks '
```

Comment (triklozoid) [reply label_studio/projects/functions/next_task.py:145]: Done.

File: label_studio/projects/functions/next_task.py (hunk 4 above)
Comment (triklozoid) [reply label_studio/projects/functions/next_task.py]: Fixed.

File: label_studio/projects/functions/next_task.py (hunk 3 above)
Comment (triklozoid) [reply label_studio/projects/functions/next_task.py:168]: 1. Added more info in the comment before this `if`
2. I think it's a problem of coverage collecting.

**Comment by makseq** [comment]
@triklozoid what will we do with project card counters? This new skip behaviour changes task numbers in queues.

**Review by makseq — DISMISSED** [review]

**Pull request merged**